rules/collapsible_heading.js

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2024 DigitalOcean
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+'use strict';
+
+/**
+ * @module modifiers/link_attributes
+ */
+
+const safeObject = require('../util/safe_object');
+
+/**
+ * @typedef {Object} CollapsibleHeadingOptions
+ * @property {string[]} [levels] List of headings to transform.
```

Comment:
Should this be an array of numbers instead, to avoid any potential confusion about whether this accepts other tokens or not?

Reply:
Changed.